java-squid/src/main/java/org/sonar/java/model/InternalSyntaxToken.java

```diff
@@ -19,42 +19,54 @@
  */
 package org.sonar.java.model;
 
-import com.google.common.collect.Lists;
-import com.sonar.sslr.api.AstNode;
-import com.sonar.sslr.api.AstNodeType;
-import com.sonar.sslr.api.GenericTokenType;
-import com.sonar.sslr.api.Token;
-import com.sonar.sslr.api.Trivia;
 import org.sonar.plugins.java.api.tree.SyntaxToken;
 import org.sonar.plugins.java.api.tree.SyntaxTrivia;
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.TreeVisitor;
+import org.sonar.sslr.grammar.GrammarRuleKey;
 
 import java.util.Iterator;
 import java.util.List;
 
 public class InternalSyntaxToken extends JavaTree implements SyntaxToken {
 
-  private final Token token;
   private List<SyntaxTrivia> trivias;
-
+  private int startIndex;
+  private int endIndex;
+  private final int line;
+  private final int column;
+  private final String value;
+  private final boolean isEOF;
 
   protected InternalSyntaxToken(InternalSyntaxToken internalSyntaxToken) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Protected method '<init>(...)' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUnusedProtectedMethod)